sklearn/metrics/_classification.py

```diff
@@ -1869,9 +1871,11 @@ def precision_recall_fscore_support(
         # score = (1 + beta**2) * precision * recall / (beta**2 * precision + recall)
         # Therefore, we can express the score in terms of confusion matrix entries as:
         # score = (1 + beta**2) * tp / ((1 + beta**2) * tp + beta**2 * fn + fp)
-        denom = beta2 * true_sum + pred_sum
+        denom = beta2 * xp.asarray(
+            true_sum, dtype=max_float_type, device=device_
+        ) + xp.asarray(pred_sum, dtype=max_float_type, device=device_)
```

Comment:
shouldn't `true_sum` and `pred_sum` already be on the device? I don't like that a simple multiplication and sum has become such a long piece of code.

Reply:
Array api strict is the only one which causes such issues. They require that the arrays be of the same type. Over here since beta2 is a float it doesn't accept ints and raises the following error:

`TypeError: array_api_strict.float64 and array_api_strict.int64 cannot be type promoted together`

Anyways I made some changes to only cast it to the correct dtype.

Reply:
Note: I verified that the CUDA tests work fine on a kaggle kernel